packages/core/src/compiler/Style.ts

```diff
@@ -608,6 +687,126 @@ const couldMatchRels = (
   return true;
 };
 
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param sebind
+ * exist as valid keys in @param subst.
+ */
+const sebindExistsInSubst = (sebind: SEBind, subst: Subst): boolean => {
+  if (sebind.contents.tag !== "StyVar") {
+    throw new Error("expected sty var");
+  }
+  const validKeywords = Object.keys(subst);
+  return validKeywords.includes(sebind.contents.contents.value);
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param expr
+ * exist as valid keys in @param subst.
+ */
+const selExprExistsInSubst = (expr: SelExpr, subst: Subst): boolean => {
+  if (expr.tag === "SEBind") {
+    return sebindExistsInSubst(expr, subst);
+  } else if (expr.tag === ("SEFunc" || "SEValCons" || "SEFuncOrValCons")) {
+    return expr.args.every((arg) => selExprExistsInSubst(arg, subst));
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param arg
+ * exist as valid keys in @param subst.
+ */
+const predArgExistsInSubst = (arg: PredArg, subst: Subst): boolean => {
+  if (arg.tag === "RelPred") {
+    return isValidRelSubst(subst, arg);
+  } else if (arg.tag === "SEBind") {
+    return sebindExistsInSubst(arg, subst);
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Determines if all the variables in @param rel exist as valid
+ * keys in @param subst.
+ *
+ * Ex, IsSubset(x,y) must have 'x' and 'y' exist as keys in @param subst
+ * for the relation to be valid.
+ */
+const isValidRelSubst = (subst: Subst, rel: RelationPattern): boolean => {
+  if (rel.tag === "RelPred") {
+    return rel.args.every((arg) => predArgExistsInSubst(arg, subst));
+  } else if (rel.tag === "RelBind") {
+    return false; // these don't have aliases
+  } else {
+    throw new Error("unknown tag");
+  }
+};
+
+/**
+ * Helper fxn for @function getRelPredAliasInstanceName
+ *
+ * Returns the substitution for a bindingform
+ */
+const getBindingFormAliasInstanceName = (
```

Comment:
`getAliasName` would be just fine. Long function names are not great for readability

Reply:
Would `getBindingFormAliasName` and `getRelPredAliasName` be ok? (I have these two as separate functions due to the recursive nature of `RelPred` and `BindingForm` types) @wodeni